apis/v1/httproute_types.go

```diff
@@ -361,6 +369,105 @@ type HTTPRouteTimeouts struct {
 	BackendRequest *Duration `json:"backendRequest,omitempty"`
 }
 
+// HTTPRouteRetry defines retry configuration for an HTTPRoute.
+//
+// Implementations SHOULD retry on connection errors (disconnect, reset, timeout,
+// TCP failure) if a retry stanza is configured.
+type HTTPRouteRetry struct {
+	// Codes defines the HTTP response status codes for which a backend request
+	// should be retried.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// <gateway:experimental>
+	Codes []HTTPRouteRetryStatusCode `json:"codes,omitempty"`
```

Comment:
If at some point we want to include strings here, do we add a new `Reasons []string` field?

Reply:
Exactly what I had been thinking if that we end up wanting that in the future!